### ISSUE TITLE
Incorporate backoff when Twitter API throws 'Too many requests'

### ISSUE THREAD

**Issue by randyzwitch** [randyzwitch/Twitter.jl#34]
Ref: #33 

~~With new changes to test suite for cursoring, Twitter API now throttles with error message of too many requests. Package works (I've tested locally), just need to figure out which test causes an issue and fix it.~~

The issue doesn't appear to be new test related, but rather than CI kicks off so many test suites simultaneously. Will turn off some Julia versions, but would also be good to have a API backoff implemented when Twitter warns to slow down.

cc: @stefanjwojcik 

https://travis-ci.org/randyzwitch/Twitter.jl/jobs/642186424

**Comment by randyzwitch** [comment]
```julia
ERROR: LoadError: HTTP.ExceptionRequest.StatusError(429, "GET", "/1.1/statuses/home_timeline.json?count=1", HTTP.Messages.Response:
"""
HTTP/1.1 429 Too Many Requests
content-length: 56
content-type: application/json;charset=utf-8
date: Tue, 28 Jan 2020 02:30:34 GMT
server: tsa_b
set-cookie: personalization_id="v1_ama+ztsWUa4hR/T8KpdE+g=="; Max-Age=63072000; Expires=Thu, 27 Jan 2022 02:30:34 GMT; Path=/; Domain=.twitter.com, guest_id=v1%3A158017863481386314; Max-Age=63072000; Expires=Thu, 27 Jan 2022 02:30:34 GMT; Path=/; Domain=.twitter.com
strict-transport-security: max-age=631138519
x-connection-hash: 06dc7fee3871af0cd2fd202bc8ed372d
x-rate-limit-limit: 15
x-rate-limit-remaining: 0
x-rate-limit-reset: 1580179247
x-response-time: 7
```

By reading the response code of `429`, then retrying after `x-rate-limit-reset`, shouldn't run into this issue again

**Comment by stefanjwojcik** [comment]
Ok, perhaps I can write a generic function for backoff and incorporate into all of these functions. I'll give that a try.

**Comment by randyzwitch** [comment]
Closed by #37
